jetbrains-core/src/software/aws/toolkits/jetbrains/core/gettingstarted/GatewaySetupAuthenticationDialog.kt

```diff
@@ -0,0 +1,190 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.core.gettingstarted
+
+import com.intellij.openapi.observable.properties.PropertyGraph
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.DialogWrapper
+import com.intellij.openapi.ui.ValidationInfo
+import com.intellij.ui.components.BrowserLink
+import com.intellij.ui.components.JBTabbedPane
+import com.intellij.util.ui.JBUI
+import com.intellij.util.ui.components.BorderLayoutPanel
+import org.jetbrains.annotations.VisibleForTesting
+import software.aws.toolkits.core.region.AwsRegion
+import software.aws.toolkits.jetbrains.core.credentials.loginSso
+import software.aws.toolkits.jetbrains.core.credentials.sono.IDENTITY_CENTER_ROLE_ACCESS_SCOPE
+import software.aws.toolkits.jetbrains.core.credentials.sono.SONO_REGION
+import software.aws.toolkits.jetbrains.core.credentials.sono.SONO_URL
+import software.aws.toolkits.jetbrains.core.gettingstarted.SetupAuthenticationNotice.NoticeType
+import software.aws.toolkits.jetbrains.core.region.AwsRegionProvider
+import software.aws.toolkits.jetbrains.utils.ui.editorNotificationCompoundBorder
+import software.aws.toolkits.resources.message
+import software.aws.toolkits.telemetry.CredentialSourceId
+import java.awt.BorderLayout
+import javax.swing.Action
+import javax.swing.BorderFactory
+import javax.swing.JComponent
+import javax.swing.JLabel
+
+data class GatewaySetupAuthenticationDialogState(
+    var idcTabState: IdentityCenterTabState = IdentityCenterTabState(),
+    val builderIdTabState: BuilderIdTabState = BuilderIdTabState,
+) {
+    private val graph = PropertyGraph()
+    val selectedTab = graph.property(GatewaySetupAuthenticationTabs.IDENTITY_CENTER)
+    data class IdentityCenterTabState(
+        var startUrl: String = "",
+        var region: AwsRegion = AwsRegionProvider.getInstance().defaultRegion(),
+        var rolePopupState: IdcRolePopupState = IdcRolePopupState()
+    )
+
+    // has no state yet
+    object BuilderIdTabState
+}
+
+enum class GatewaySetupAuthenticationTabs {
+    IDENTITY_CENTER,
+    BUILDER_ID
+}
+
+class GatewaySetupAuthenticationDialog(
+    private val project: Project?,
+    private val scopes: List<String> = emptyList(),
+    private val state: GatewaySetupAuthenticationDialogState = GatewaySetupAuthenticationDialogState(),
+    private val tabSettings: Map<GatewaySetupAuthenticationTabs, AuthenticationTabSettings> = emptyMap(),
+    private val promptForIdcPermissionSet: Boolean = false,
+) : DialogWrapper(project), AuthenticationDialog {
+    private val rootTabPane = JBTabbedPane()
+    private val idcTab = IdcTabPanelBuilder(state.idcTabState::startUrl, state.idcTabState::region).build()
+    private val builderIdTab = BuilderIdTabPanelBuilder().build()
+    private val wrappers = GatewaySetupAuthenticationTabs.values().associateWith { BorderLayoutPanel() }
+    var attempts = 0
+        private set
```

Comment:
not seeing where this ends up getting incremented

Reply:
The handlers for this dialog were copied from the builderId handler in SetupAuthenticationDialog which also does not touch the attempts variable since it uses loginSso under the hood. I need some help to understand how I can update both dialogs to report unsuccessful attempts.